code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+		Vec3 vecU = vecT - LUV::ProjLine( vecT, vecM, vecP );
+		dirU.push_back( vecU / LUV::Length( vecU ) );
+		Vec3 vecA = LUV::ProjPlane( vecR, vecM, LUV::PlaneNormal( vecM, vecP, vecT ) );
+		Vec3 vecB = LUV::ProjLine( vecA, vecM, vecP );
+		obsTet.push_back( ObservationTetrahedron( vecR, vecA, vecB, vecP ) );
+		obsTet.push_back( ObservationTetrahedron( vecR, vecA, vecB, vecM ) );
+	}
+	
+	void AddFace( const Vec3& vecA, const Vec3& vecB, const Vec3& vecC, const Vec3& vecU , const Vec3& vecR )
+	{
+		Vec3 vecN = vecU - LUV::ProjPlane( vecU, vecA, LUV::PlaneNormal( vecA, vecB, vecC ) );
+		dirN.push_back( vecN / LUV::Length( vecN ) );
+		AddEdge( vecA, vecB, vecC, vecR );
+		AddEdge( vecB, vecC, vecA, vecR );
+		AddEdge( vecC, vecA, vecB, vecR );
+	}
+	
+public:
+	
+	SphereTetrahedronIntersection( const Vec3& vecTetA, const Vec3& vecTetB, const Vec3& vecTetC, const Vec3& vecTetD, const Vec3& vecSphCenter )
+	{
+		AddFace( vecTetA, vecTetB, vecTetC, vecTetD, vecSphCenter );
+		AddFace( vecTetB, vecTetC, vecTetD, vecTetA, vecSphCenter );
+		AddFace( vecTetC, vecTetD, vecTetA, vecTetB, vecSphCenter );
+		AddFace( vecTetD, vecTetA, vecTetB, vecTetC, vecSphCenter );
+		for( int idf = 0; idf < 4; ++idf )
+		{
+			for( int idl = 0; idl < 3; ++idl )
+			{
+				int ids = 3 * idf + idl;
+				int idp = 2 * ids;
+				int idm = idp + 1;
+				
+				obsTetSgn.push_back(
+					LUV::Dot( obsTet[idp].prdA, dirN[idf] )
+					* LUV::Dot( obsTet[idp].prdF, dirL[ids] )
+					* LUV::Dot( obsTet[idp].prdD, dirU[ids] )
+				);
+				
+				obsTetSgn.push_back(
+					- LUV::Dot( obsTet[idm].prdA, dirN[idf] )
+					* LUV::Dot( obsTet[idm].prdF, dirL[ids] )
+					* LUV::Dot( obsTet[idm].prdD, dirU[ids] )
+				);
+			}
+		}
+	}
+	
+	double GetSolidAngle( double radius )
+	{
+		double solidAngle = 0;
+		for( int idx = 0; idx < 24; ++idx )
+		{
+			solidAngle += obsTet[idx].GetSolidAngle( radius ) * obsTetSgn[idx];
+		}
+		return solidAngle;
+	}
+	
+	double GetSurfaceArea( double radius )
+	{
+		return GetSolidAngle( radius ) * radius * radius;
+	}
+	
+	double GetVolume( double radius )
+	{
+		double volume = 0;
+		for( int idx = 0; idx < 24; ++idx )
+		{
+			volume += obsTet[idx].GetVolume( radius ) * obsTetSgn[idx];
+		}
+		return volume;
+	}
+};
+
+int main()
+{
+    using namespace std;
+    
+    // An example tetrahedron with a volume of 1000
+    Vec3 vecTetA( 0, 0, 0 );
+    Vec3 vecTetB( 10, 0, 0 );
+    Vec3 vecTetC( 0, 20, 0 );
+    Vec3 vecTetD( 0, 0, 30 );
+    
+    // Center of the sphereş
+    Vec3 vecSphCenter( 3, 3, 3 );
+    
+    SphereTetrahedronIntersection testIntersection( vecTetA, vecTetB, vecTetC, vecTetD, vecSphCenter );
+    
+    // @ radius = 1 
+    double srf = testIntersection.GetSurfaceArea( 1 ) / pi; // must be 4
+    double vol = testIntersection.GetVolume( 1 ) / pi; // must be 4/3
+    cout << srf << ", " << vol << endl;
+    
+    // @ radius = 30
+    srf = testIntersection.GetSurfaceArea( 30 ); // must be 0
```

Comment:
Why did you reassign the variables? Couldn't you have just printed what the function returns? This way, you could get rid of the variables altogether and just print out the values above as well.

Reply:
For no reason. It's just a simple test of the functionality, I haven't given much thought about it.

I might have unconsciously thought repeating this in a single line might make it uncomfortably long to read:
"testIntersection.GetSurfaceArea(...)"